### ISSUE TITLE
Loops only work once

### ISSUE THREAD

**Issue by michaelgaultjr** [GMDU/code-of-copper#32]
I've run into an issue where if a script has multiple loops, the first loop while execute, but execution will stop at any subsequent loops. However `while` loops continue to work

Script using the `loop` keyword (broken)
```
# Area Miner
# direction controls
let x = 0
let z = -1
let size = 5

# opposite directions
# using * -1 as using -x or -z resulted in 0 instead of 1 or -1
let ix = x * -1
let iz = z * -1

loop size
  place 0, [x, 0, z]
  move [x, 0, z]
end

print "Finished first loop"

place 0, [z, 0, x]
move [z, 0, x]

print "Final line executed"

loop size 
  place 0, [ix, 0, iz]
  move [ix, 0, iz]
end
print "This doesn't execute"
```

Script using the `while` keyword (works)
```
# Area Miner
# direction controls
let x = 0
let z = -1
let size = 5

# opposite directions
# using * -1 as using -x or -z resulted in 0 instead of 1 or -1
let ix = x * -1
let iz = z * -1

let i = 0
while i < size
  place 0, [x, 0, z]
  move [x, 0, z]
  let i = i + 1
end

print "Finished first loop"

place 0, [z, 0, x]
move [z, 0, x]

print "Final line executed"

let i = 0
while i < size
  place 0, [ix, 0, iz]
  move [ix, 0, iz]
  let i = i + 1
end
print "This executes"
```

**Comment by moxvallix** [comment]
I'll have a look into this, but it might already be fixed in v0.3.x, because there was a major refactor there.

In any case, v0.2.x will not receive further updates, so this will be fixed in v0.3.x.

**Comment by moxvallix** [comment]
@GearsDatapacks I have confirmed this is still an issue in v0.3.x; my guess is that the loop instruction actually modifies the value of the variable, meaning `size` is zero by the time of the second loop.
